File: 括号匹配深度.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    string s;
    cin >> s;
    int ans = 0, cnt = 0;
    for (char c : s) {
        if (c == '(') {
            cnt += 1;
            ans = max(ans, cnt);
        } else {
            cnt -= 1;
        }
    }
    cout << ans << endl;
    return 0;
}

File: 括号匹配深度.py
import sys
input = lambda: sys.stdin.readline().strip()
mod = 10 ** 9 + 7 # 998244353

s = list(input())
ans, cnt = 0, 0
for c in s:
    if c == '(':
        cnt += 1
        ans = max(ans, cnt)
    else:
        cnt -= 1
print(ans)
